Options to configure start folder index, hide previous results and new title bar when using regex matches

carlosflorencio opened the pull request:
Hey 👋 

Adding 3 new small improvements:
- New option `startFolderDisplayIndex` that allows to customise the first folder shown in the results. 
  - For my taste, I prefer to start at index `1` instead of `0` because it's a bit pointless to see the the workspace folder taking up space on every item. And since I work with monorepos, the folder at index `1` is more useful for me because it's the app name. I've kept the default as `0` since it's the current behaviour.
 - New option `showPreviousResultsWhenNoMatches` to clear the results if there are no matches for the current query. Defaults to `true` since it's the current behaviour.
   - I prefer to have strong visual indication (no results instead of vscode no bold match the results) when there are no matches for the current query, I then know that I should tweak my query.
 - New functionality to show the QuickPick title bar when using the regex param matches. When there is a match, we show a 'preview' of the `rg` expanded params in the title. 

## Demo

https://github.com/joshmu/periscope/assets/1500881/95d2c4bc-ad07-4353-b2af-f5766f287b6b

In the video I'm hiding the results when the query doesn't match anything. You can see the title bar appearing when there is a regex match.

joshmu commented:
Hey @carlosflorencio 👋 

These are all excellent ideas, and I'm happy you are finding utility in the extension.

1. `startFolderDisplayIndex`
Yep we can include this, I use multiple repo folders in the one workspace so index 0 was for my use case.

2.  `showPreviousResultsWhenNoMatches`
Keep the option there but let's set this to `false` by default. I think you are right and provides more transparency when a command is no longer valid :)

3. **show query param rg command**
Love this, happy to have this enabled by default, would be good to expose an option to disable this.

joshmu reviewed (changes requested):
Hey @carlosflorencio 👋 

These are all excellent ideas, and I'm happy you are finding utility in the extension.

1. `startFolderDisplayIndex`
Yep we can include this, I use multiple repo folders in the one workspace so index 0 was for my use case.

2.  `showPreviousResultsWhenNoMatches`
Keep the option there but let's set this to `false` by default. I think you are right, it provides more transparency when a command is no longer valid :)

3. **show query param rg command**
Love this, happy to have this enabled by default, would be good to expose an option to disable this.

carlosflorencio commented:
@joshmu I've addressed your comments in the last two pushed commits.

- Updated the setting `showPreviousResultsWhenNoMatches` default to `false`
- Added a new setting `rgQueryParamsShowTitle` to disable the title update

Happy Holidays :)

joshmu commented:
@carlosflorencio Merged! Happy holidays to you too!
